vispy/visuals/volume.py

```diff
@@ -224,6 +231,36 @@
     return linePosition + ( scaleFactor * lineVector );
 }
 
+vec3 intersectLineSphere(vec3 origin,
+                         vec3 direction,
+                         vec3 center,
+                         float radius,
+                         out bool hit_sphere) {
+    vec3 oc = origin - center;
+    // a = 1 because direction is normalized
+    float b = 2 * dot(oc, direction);
+    float c = dot(oc, oc) - radius * radius;
+    float discriminant = b * b - 4 * c;
+    if (discriminant < 0) {
+        // no intersection
+        hit_sphere = false;
+        return vec3(0.0);
+    }
+    float t0 = (-b - sqrt(discriminant)) / 2.0;
+    float t1 = (-b + sqrt(discriminant)) / 2.0 ;
+    if (t0 <= 0.0 && t1 <= 0.0) {
+        // both intersections are behind the camera
+        hit_sphere = false;
+        return vec3(0.0);
+    }
+    // it's okay one of the intersections is behind the camera
+    // we still want the depth from the "front" of the sphere
+    hit_sphere = true;
+    return origin + min(t0, t1) * direction;
+}
+
+$def_tf
```

Comment:
In principle this is not really needed? You can just use `$apply_transfer_function` in the place where it's needed and the definition will be placed where appropriate... though it might not work if the snippet has multiple functions inside... It's fine like this if this gets too tricky :P But for reference, look at how `get_data` is defined.

Reply:
I admit I don't understand all the templating options.

I'm not sure the method you're suggesting would allow you to do things like add new uniforms. Or, like you mention, multiple functions.Maybe these are not such important features though - you could always re-define an update the instance. Maybe that's not quite as performant.

Reply:
Yeah, definitely not blocking :)